supplier_api_tester/supplier_api_tester/tests/booking.py

```diff
@@ -141,3 +141,57 @@ def test_booking(api_url, api_key, product_id, timeslots: bool, version=1):
                 )
 
     return TestResult()
+
+@test_wrapper
+def test_cancellation(api_url, api_key, product_id, timeslots: bool, version=1):
+    '''Perform booking that will be cancelled'''
+    url = f'{api_url}/v{version}/products/{product_id}/reservation'
+    slot = get_reservation_slot(api_url, api_key, product_id, timeslots)
+    variant_quantity_map = {
+         variant.id: 2 for variant in slot.variants if variant.max_tickets > 2
+    }
+
+    json_payload = get_payload_from_slot(slot, variant_quantity=2, min_quantity=3)
+    if timeslots:
+        json_payload['timeslot'] = slot.start
+    raw_response, response = client(url, api_key, method=requests.post, json_payload=json_payload)
+    reservation = get_reservation(raw_response, response)
+
+    url = f'{api_url}/v{version}/booking'
+    raw_response, response = client(url, api_key, method=requests.post, json_payload={
+        'reservation_id': reservation.reservation_id,
+    })
+    booking = get_booking(raw_response, response)
+    booking_id = booking.booking_id
+
+    # cancel existing booking
+    url = f'{api_url}/v{version}/booking/{booking_id}'
+    raw_response, response = client(url, api_key, method=requests.delete, json_payload={"booking_id": booking_id})
+    if raw_response.status_code == 204:
+        booking.is_cancelled = True
+
+    # cancel booking with no ID/non-existent ID
+    non_existing_booking_id = "I-DO-NOT-EXIST"
+    url = f'{api_url}/v{version}/booking/{non_existing_booking_id}'
+    raw_response, response = client(url, api_key, method=requests.delete, json_payload={"booking_id": non_existing_booking_id})
+    api_error = get_api_error(raw_response, response)
+    expected_error = ApiError(
+        error_code=1004,
+        error='Missing booking',
+        message='Required argument \'booking_id\' was not found',
+    )
+    check_api_error(raw_response, api_error, expected_error)
+
+    # cancel booking that was already cancelled:
```

Comment:
Place this code right after the first cancellation so there will be no need for setting the flag (`booking.is_cancelled = True`).